azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositePushNotificationInfo.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.models;
+
+import com.azure.android.communication.common.CommunicationIdentifier;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * CallCompositePushNotificationInfo for forwarding calling push notifications to UI Library.
+ */
+public class CallCompositePushNotificationInfo {
+    private final Map<String, String> notificationInfoMap;
+    private final String fromDisplayName;
+    private final String from;
+    private final boolean isIncomingWithVideo;
+    private final String callId;
+    private final String to;
+    private final String eventTypeResult;
+
+    /**
+     * Create {@link CallCompositePushNotificationInfo}.
+     * @param notificationInfoMap Notification info map.
+     */
+    public CallCompositePushNotificationInfo(final Map<String, String> notificationInfoMap)
+            throws ClassNotFoundException, NoSuchMethodException,
+            InvocationTargetException, IllegalAccessException, InstantiationException {
+
+        final Class<?> pushNotificationInfoClass =
```

Comment:
Why the reflection?  Do we not have access to this class?

Reflection means a lot of code safety loss, evidenced by all the checked exceptions.

We should consume the checked exceptions in here. Emit a new exception if there is a fatal error. A consumer of this API can't do anything to recover from any of those exceptions outside this method, so this is the best place to handle them.